aerogear/app-metrics-operator
# Add Monitor for Operator + Apply Standards

## Motivation
- Apply the same standard used in the other projects as UPS Oper in order to add the Monitoring features easily. (https://issues.jboss.org/browse/AEROGEAR-9598)
- https://issues.jboss.org/browse/AEROGEAR-9190
- https://issues.jboss.org/browse/AEROGEAR-9189

## What
- Add default prometheus and graphana dashboard for the operator. 
- Add GitHub files from standards ( same applied in all projects ) 
- Same Makefile structure of UPS Oper
- Same README structure of UPS Oper
- Add unit test, coveralls and CI as UPS Oper
- Add release process as UPS Oper
- Add e2e test as it was done for UPS Oper
- Improve APP NAMESPACE and OPER namespace validation in order to allow unit test and run it locally + unit tests to ensure that it is working as it was done for other projects. 

## Why
- Be able to easily create the monitor command and apply changes horizontally
- Follow the standards and make easier to keep maintained
- Add requirements to create the SOP's and Monitor features  

## Steps to verify
- **Check the README changes;** ( same ups oper structure )
- **Check that the CI working:**
- **Check that the Jenkins is working here with the e2e test as well now**.
- Check the Graphana dash here: https://grafana-route-middleware-monitoring.apps.london-2f09.openshiftworkshop.com/d/null/app-metrics-operator?refresh=10s&orgId=1
- Check the Prometheus Rules/Alerts here: https://prometheus-route-middleware-monitoring.apps.london-2f09.openshiftworkshop.com/alerts

<img width="1666" alt="Screenshot 2019-07-16 at 07 03 23" src="https://user-images.githubusercontent.com/7708031/61285746-df506900-a797-11e9-93f4-ce407037dbbc.png">

<img width="1047" alt="Screenshot 2019-07-16 at 07 02 20" src="https://user-images.githubusercontent.com/7708031/61285654-b5974200-a797-11e9-8f37-68338962398f.png">

<img width="1026" alt="Screenshot 2019-07-15 at 16 25 52" src="https://user-images.githubusercontent.com/7708031/61242917-41b85380-a71d-11e9-80fa-859ffeed5138.png">


## Checklist:

## Progress
